### PR TITLE
Eval type fix in V1

### PR DESCRIPTION
**Pull request by sydney-runkle**
Fix https://github.com/pydantic/pydantic/issues/9613 (we'll have to pull this into V2 with a release to fix this on the V2 side)

Skip change file check

### CODE REVIEW

**Comment by sydney-runkle** [comment]
Fixes this issue:

```
tests/test_v1.py:2: in <module>
    from pydantic.v1 import VERSION as V1_VERSION
pydantic/v1/__init__.py:7: in <module>
    from pydantic.v1.env_settings import BaseSettings
pydantic/v1/env_settings.py:23: in <module>
    class BaseSettings(BaseModel):
pydantic/v1/main.py:178: in __new__
    annotations = resolve_annotations(namespace.get('__annotations__', {}), namespace.get('__module__', None))
pydantic/v1/typing.py:402: in resolve_annotations
    value = _eval_type(value, base_globals, None)
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/typing.py:471: in _eval_type
    _deprecation_warning_for_no_type_params_passed("typing._eval_type")
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/typing.py:451: in _deprecation_warning_for_no_type_params_passed
    warnings.warn(depr_message, category=DeprecationWarning, stacklevel=3)
E   DeprecationWarning: Failing to pass a value to the 'type_params' parameter of 'typing._eval_type' is deprecated, as it leads to incorrect behaviour when calling typing._eval_type on a stringified annotation that references a PEP 695 type parameter. It will be disallowed in Python 3.15.
```

File: pydantic/typing.py [hunk 1]
```diff
@@ -399,7 +399,10 @@ def resolve_annotations(raw_annotations: Dict[str, Type[Any]], module_name: Opti
             else:
                 value = ForwardRef(value, is_argument=False)
         try:
-            value = _eval_type(value, base_globals, None)
+            if sys.version_info > (3, 13):
```
Comment (hramezani) [inline pydantic/typing.py:402]: Shouldn't it be `>= 3.13`?
```suggestion
            if sys.version_info >= (3, 13):
```

Comment (alexmojaki) [inline pydantic/typing.py:402]:
```suggestion
            if sys.version_info[:2] >= (3, 13):
```

File: pydantic/typing.py [hunk 2]
```diff
@@ -399,7 +399,10 @@ def resolve_annotations(raw_annotations: Dict[str, Type[Any]], module_name: Opti
...
+                value = _eval_type(value, base_globals, None, type_params=())
```
Comment (alexmojaki) [inline pydantic/typing.py:403]: I assume we're fine with the new type param syntax just not working in v1?

File: pydantic/typing.py (hunk 1 above)
Comment (sydney-runkle) [reply pydantic/typing.py]: I don't think I think we use the pattern I've suggested for almost all of our version info checks...

File: pydantic/typing.py (hunk 2 above)
Comment (sydney-runkle) [reply pydantic/typing.py:403]: Yep, correct!

Comment (sydney-runkle) [reply pydantic/typing.py:403]: @vfazio, to your point from earlier, this is more of a bandaid and not an actual fix (we're not really adding support for 3.13, just ensuring it doesn't break things)

**Comment by sydney-runkle** [comment]
We're in no rush for a release on the V1 front here, so I just cherry picked this over to the v1 namespace in V2, but we can include this in the next v1 patch release for good measure + to make the v1 update process from within v2 as smooth as possible.

**Pull request merged**